packages/ilios-common/addon/components/leadership-expanded.js

```diff
@@ -18,31 +18,61 @@ export default class CourseLeadershipExpandedComponent extends Component {
   }
 
   @cached
-  get courseDirectors() {
-    return new TrackedAsyncData(this.args.course.directors);
+  get modelHasDirectors() {
+    return 'directors' in this.args.model;
   }
 
   @cached
-  get courseAdministrators() {
-    return new TrackedAsyncData(this.args.course.administrators);
+  get modelHasAdministrators() {
+    return 'administrators' in this.args.model;
   }
 
   @cached
-  get courseStudentAdvisors() {
-    return new TrackedAsyncData(this.args.course.studentAdvisors);
+  get modelHasStudentAdvisors() {
+    return 'studentAdvisors' in this.args.model;
+  }
+
+  @cached
+  get modelDirectors() {
+    if (this.modelHasDirectors) {
+      return new TrackedAsyncData(this.args.model.directors);
+    }
+    return null;
+  }
+
+  @cached
+  get modelAdministrators() {
+    if (this.modelHasAdministrators) {
+      return new TrackedAsyncData(this.args.model.administrators);
+    }
+    return null;
+  }
+
+  @cached
+  get modelStudentAdvisors() {
+    if (this.modelHasStudentAdvisors) {
+      return new TrackedAsyncData(this.args.model.studentAdvisors);
+    }
+    return null;
   }
 
   get directors() {
-    const directors = this.courseDirectors.isResolved ? this.courseDirectors.value.slice() : [];
+    if (!this.modelHasDirectors) {
+      return [];
+    }
+    const directors = this.modelDirectors.isResolved ? this.modelDirectors.value.slice() : [];
```

Comment:
I'm not sure these need to get sliced. IIRC in Ember data these are arrays now or at least array proxies and don't need special handling anymore.

Reply:
nice catch. removed. we should do that everywhere then.

Reply:
:heavy_check_mark: